packages/shared-ux/page/analytics_no_data/impl/src/analytics_no_data_page.component.tsx

```diff
@@ -21,26 +23,63 @@ export interface Props {
   allowAdHocDataView?: boolean;
   /** if the kibana instance is customly branded */
   showPlainSpinner: boolean;
+  /** The flavor of the empty page to use. */
+  pageFlavor?: AnalyticsNoDataPageFlavor;
+  prependBasePath: (path: string) => string;
 }
 
-const solution = i18n.translate('sharedUXPackages.noDataConfig.analytics', {
-  defaultMessage: 'Analytics',
-});
-
-const pageTitle = i18n.translate('sharedUXPackages.noDataConfig.analyticsPageTitle', {
-  defaultMessage: 'Welcome to Analytics!',
-});
-
-const addIntegrationsTitle = i18n.translate('sharedUXPackages.noDataConfig.addIntegrationsTitle', {
-  defaultMessage: 'Add integrations',
-});
-
-const addIntegrationsDescription = i18n.translate(
-  'sharedUXPackages.noDataConfig.addIntegrationsDescription',
-  {
-    defaultMessage: 'Use Elastic Agent to collect data and build out Analytics solutions.',
-  }
-);
+const flavors: {
+  [K in AnalyticsNoDataPageFlavor]: (deps: {
+    kibanaGuideDocLink: string;
+    prependBasePath: (path: string) => string;
+  }) => KibanaNoDataPageProps['noDataConfig'];
+} = {
+  kibana: ({ kibanaGuideDocLink }) => ({
+    solution: i18n.translate('sharedUXPackages.noDataConfig.analytics', {
+      defaultMessage: 'Analytics',
+    }),
+    pageTitle: i18n.translate('sharedUXPackages.noDataConfig.analyticsPageTitle', {
+      defaultMessage: 'Welcome to Analytics!',
+    }),
+    logo: 'logoKibana',
+    action: {
+      elasticAgent: {
+        title: i18n.translate('sharedUXPackages.noDataConfig.addIntegrationsTitle', {
+          defaultMessage: 'Add integrations',
+        }),
+        description: i18n.translate('sharedUXPackages.noDataConfig.addIntegrationsDescription', {
+          defaultMessage: 'Use Elastic Agent to collect data and build out Analytics solutions.',
+        }),
+        'data-test-subj': 'kbnOverviewAddIntegrations',
+      },
+    },
+    docsLink: kibanaGuideDocLink,
+  }),
+  serverless_search: ({ prependBasePath }) => ({
+    solution: i18n.translate('sharedUXPackages.noDataConfig.analytics', {
+      defaultMessage: 'Elasticsearch',
+    }),
+    pageTitle: i18n.translate('sharedUXPackages.noDataConfig.analyticsPageTitle', {
+      defaultMessage: 'Welcome to Elasticsearch!',
+    }),
+    logo: 'logoElasticsearch',
+    action: {
+      elasticsearch: {
+        title: i18n.translate('sharedUXPackages.noDataConfig.elasticsearchTitle', {
+          defaultMessage: 'Get started',
+        }),
+        description: i18n.translate('sharedUXPackages.noDataConfig.elasticsearchDescription', {
+          defaultMessage:
+            'Set up your programming language client, ingest some data, and start searching.',
+        }),
+        'data-test-subj': 'kbnOverviewElasticsearchGettingStarted',
+        href: prependBasePath('/app/elasticsearch/'),
+        /** force the no data card to be shown **/
+        canAccessFleet: true,
```

Comment:
Passing/faking this explicitly so that `NoDataCard` doesn't fallback to the "no access to fleet" page. This needs to be refactored. 

Currently, it is ugly: even though the `NoDataCard` seem to be generic and text and links can be configured through props, some "fleet" logic parts are still hardcoded inside the component. For example, the component checks the access to fleet using context.